tests/bud.bats

```diff
@@ -202,7 +202,111 @@ _EOF
   expect_output --substring $targetarch
 }
 
-# Test pinning image using additional build context
+# Test skipping images with FROM
+@test "build-test skipping unwanted stages with FROM" {
+  mkdir -p ${TEST_SCRATCH_DIR}/bud/platform
+
+  cat > ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile << _EOF
+FROM alpine
+RUN echo "unwanted stage"
+
+FROM alpine as one
+RUN echo "needed stage"
+
+FROM alpine
+RUN echo "another unwanted stage"
+
+FROM one
+RUN echo "target stage"
+_EOF
+
+  run_buildah build $WITH_POLICY_JSON -t source -f ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile
+  expect_output --substring "needed stage"
+  expect_output --substring "target stage"
+  assert "$output" !~ "unwanted stage"
+}
+
+# Test skipping unwanted stage with COPY from stage name
+@test "build-test skipping unwanted stages with COPY from stage name" {
+  mkdir -p ${TEST_SCRATCH_DIR}/bud/platform
+
+  cat > ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile << _EOF
+FROM alpine
+RUN echo "unwanted stage"
+
+FROM alpine as one
+RUN echo "needed stage"
+RUN echo hey > hey
+
+FROM alpine
+COPY --from=one hey .
+RUN cat hey
+RUN echo "target stage"
+_EOF
+
+  run_buildah build $WITH_POLICY_JSON -t source -f ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile
+  expect_output --substring "needed stage"
+  expect_output --substring "hey"
```

Comment:
What part of the build output is this trying to match?  That string appears multiple times, in instructions in multiple stages.